ui/app/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -0,0 +1,330 @@
+            >
+              {this.context.t('decryptMetamask')}
+            </div>
+          </div>
+        </div>
+        { hasDecrypted ?
+          (
+            <div
+              className={classnames({
+                'request-decrypt-message__message-copy': true,
+                'request-decrypt-message__message-copy--pressed': copyToClipboardPressed,
+              })}
+              onClick={() => this.copyMessage()}
+              onMouseDown={() => this.setState({ copyToClipboardPressed: true })}
+              onMouseUp={() => this.setState({ copyToClipboardPressed: false })}
+            >
+              <Tooltip
+                position="bottom"
+                title={hasCopied ? this.context.t('copiedExclamation') : this.context.t('copyToClipboard')}
+                wrapperClassName="request-decrypt-message__message-copy-tooltip"
+              >
+                <div
+                  className="request-decrypt-message__message-copy-text"
+                >
+                  {this.context.t('decryptCopy')}
+                </div>
+                <img src="images/copy-to-clipboard.svg" />
+              </Tooltip>
+            </div>
+          )
+          :
+          <div></div>
+        }
+      </div>
+    )
+  }
+
+  renderFooter = () => {
+    const { txData } = this.props
+
+    return (
+      <div className="request-decrypt-message__footer">
+        <Button
+          type="default"
+          large
+          className="request-decrypt-message__footer__cancel-button"
+          onClick={async event => {
+            this._removeBeforeUnload()
+            await this.props.cancelDecryptMessage(txData, event)
+            this.context.metricsEvent({
+              eventOpts: {
+                category: 'Transactions',
+                action: 'Decrypt Message Request',
+                name: 'Cancel',
+              },
+            })
+            this.props.clearConfirmTransaction()
+            this.props.history.push(DEFAULT_ROUTE)
+          }}
+        >
+          { this.context.t('cancel') }
+        </Button>
+        <Button
+          type="secondary"
+          large
+          className="request-decrypt-message__footer__sign-button"
+          onClick={async event => {
+            this._removeBeforeUnload()
+            await this.props.decryptMessage(txData, event)
+            this.context.metricsEvent({
+              eventOpts: {
+                category: 'Transactions',
+                action: 'Decrypt Message Request',
+                name: 'Confirm',
+              },
+            })
+            this.props.clearConfirmTransaction()
+            this.props.history.push(DEFAULT_ROUTE)
+          }}
+        >
+          { this.context.t('decrypt') }
+        </Button>
+      </div>
+    )
+  }
+
+  render = () => {
+    return (
+      <div className="request-decrypt-message__container">
+        { this.renderHeader() }
```

Comment:
So the key difference between this current version and the most recent designs is the use of the `ConfirmTransactionBase` wrapper component (and the header and footer that are used by that). Removing this header and footer, and passing in `renderBody()` to `ConfirmTransactionBase` as the `contentComponent` prop should make the header and footer conform to the designs. Take a look at the `ui/app/pages/confirm-approve/confirm-approve.component.js` for an example of how to do this.

The same goes for `confirm-encryption-public-key.component.js` as well - it should use `ConfirmTransactionBase` instead of these custom headers/footers.

Reply:
We can't use `ConfirmTransactionBase`, because we don't need to have a gas estimator and other components for such dialog. Also such component very similar on `personal_sign` by design. I don't sure that we need to have different designs for similar operations.

Reply:
Ah I see - I hadn't realized that `ConfirmTransactionBase` included transaction costs and gas estimations. You're right, that doesn't make sense. Hmm.